File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map < Integer, Integer> difference= new HashMap <>();
        for ( int i=0; i<nums.length;i++){
           Integer diffIndex = difference.get(nums[i]);
           if( diffIndex != null){
               return new int[]{diffIndex,i};
           }
           else{
               difference.put(target-nums[i],i);
           }
        }
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        valueMap = {}

        for i,n in enumerate(nums):

            diff = target - n 
            if diff in valueMap:
                return [valueMap[diff],i]
            
            valueMap[n] = i
        
    